Logging pages for Elixir and Node (#859)
appsignal/appsignal-docs: source/posts/logging/platforms/integrations/elixir.mdx

</Notice>

To use logging with your AppSignal integration for Elixir, you first need to install version `2.4` of the package or higher.

## Configure Logging

<Notice variant="warning">
  ⚠️ Log data must not contain personal data, such as names, email addresses,
  etc. It is your responsibility to ensure this data is filtered out before
  being sent to AppSignal, and when identifying a person is necessary that your
  application uses alternative forms of identification, such as a user ID, hash,
  or pseudonym.
</Notice>

### Usage

Once the logging branch has been successfully installed, your Elixir integration will include a new `Appsignal.Logger` module, which provides eight functions.

Each of these functions correspond to the severity of the log message:

- `debug/3`,
- `info/3`,
- `notice/3`,
- `warning/3`,
- `error/3`,
- `critical/3`,
- `alert/3`,
- `emergency/3`

These functions accept three arguments in the following order: `group`, `message`, and optionally `metadata`. The examples below show how you can use these arguments in your application when creating new logs:

```elixir
Appsignal.debug("app", "Log message line")

> Appsignal.debug("app", "Something went wrong with customer: #{customer_id}", %{customer_id: 123, post_id: 234})

Reviewer: Hm, shouldn't this be `Appsignal.Logger.debug()`?

Author: Yes! Thanks for fixing it.

> Appsignal.debug("app", "Something went wrong with customer: #{customer_id}", %{customer_id: 123, post_id: 234})

Reviewer 2: ```suggestion
Appsignal.debug("app", "Log message line")
Appsignal.debug("app", "Something went wrong with customer: #{customer_id}", %{customer_id: 123, post_id: 234})
```

> Appsignal.debug("app", "Something went wrong with customer: #{customer_id}", %{customer_id: 123, post_id: 234})

Reviewer 2: Is it necessary to also include the customer_id in the message if it's also part of the attributes? The Node.js example doesn't do this.

Author: It's not necessary, it's just an example.

---
title: "Logging From Elixir"
sup: "BETA"
---

<Notice>

🧪 AppSignal Logging is currently only available to Beta Testers. If you think your
application is a good candidate, [send us an
email](mailto:support@appsignal.com?subject=Logging%20beta)!

</Notice>

## Configure Package

<Notice variant="warning">

This feature is designed to work from AppSignal for Elixir integration v2 and greater.

</Notice>

To use logging with your AppSignal integration for Elixir, you first need to install the `logging-feature` branch of the AppSignal package. To do this, open your application's `mix.exs` file, and find the `appsignal` package:

```elixir
{:appsignal, #...}
```

and replace the line with the following code:

```elixir

> {:appsignal, git: "https://github.com/appsignal/appsignal-elixir", branch: "logging-feature"}

Reviewer: Since we have released new versions with the logging feature, let's remove the references to specific branches.

```

After altering and saving your mix file, run `$ mix deps.get` to install the logging version of the package.

## Configure Logging

<Notice variant="warning">
  ⚠️ Log data must not contain personal data, such as names, email addresses,
  etc. It is your responsibility to ensure this data is filtered out before
  being sent to AppSignal, and when identifying a person is necessary that your
  application uses alternative forms of identification, such as a user ID, hash,
  or pseudonym.
</Notice>

### Usage

> There's a new module exposed in the integration: `Appsignal.Logger`. It provides eight functions:

Reviewer 3: ```suggestion
Once the logging branch has been successfully installed, your Elixir integration will include a new `Appsignal.Logger` module, which provides eight functions. 

Each of these functions correspond to the severity of the log message:
```

- `debug/3`,
- `info/3`,
- `notice/3`,
- `warning/3`,
- `error/3`,
- `critical/3`,
- `alert/3`,
- `emergency/3`

> each one of them corresponding to the severity of the log message.
> 
> The first two arguments are the `group` and the `message`. The third is an optional one for `metadata`.

Reviewer 3: ```suggestion
These functions accept three arguments in the following order: `group`, `message`, and optionally `metadata`. The examples below show how you can use these arguments in your application when creating new logs:
```
